torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1246,6 +1246,35 @@ int computeDeltaMS(
       .count();
 }
 
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutErrorMsg(
+    const std::string& extraMsg) {
+  return c10::str(
+      logPrefix(),
+      "Received a dump signal due to a collective timeout from ",
+      extraMsg,
+      " and will ",
+      "start to dump the debug info. ",
+      "Last enqueued NCCL work: ",
+      pgStatus_->lastEnqueuedSeq,
+      ", last completed NCCL work: ",
+      pgStatus_->lastCompletedSeq,
+      ".");
+}
+
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutExitMsg(
+    const std::string& extraMsg) {
+  return c10::str(
+      "ProcessGroupNCCL's watchdog detected an exception from ",
+      extraMsg,
+      ". ",
+      "This is most likely caused by incorrect usages of collectives, e.g., wrong ",
```

Comment:
following "do not repeat yourself" principle of sw eng..  

why do we need to have 2 very similar messages at all? Can we think this through and find a way to only have one message?  (or if we want 2, can we make the first one more detailed and the second one at exit smaller?

Reply:
I see. I think historically we have an exit message when terminate or exit Monitoring thread. We did have a long exit message for watchdog hang.

Reply:
I like the idea of making 2 simple.